packages/graphql/src/translate/where/utils.ts

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { PointField, PrimitiveField } from "../../types";
+
+export type NumericalWhereOperator = "GT" | "GTE" | "LT" | "LTE";
+export type SpatialWhereOperator = "DISTANCE";
+export type StringWhereOperator = "CONTAINS" | "STARTS_WITH" | "ENDS_WITH";
+export type RegexWhereOperator = "MATCHES";
+export type ArrayWhereOperator = "IN" | "INCLUDES";
+export type RelationshipWhereOperator = "ALL" | "NONE" | "SINGLE" | "SOME";
+
+export type WhereOperator =
+    | "NOT"
+    | NumericalWhereOperator
+    | SpatialWhereOperator
+    | StringWhereOperator
+    | `NOT_${StringWhereOperator}`
+    | RegexWhereOperator
+    | ArrayWhereOperator
+    | `NOT_${ArrayWhereOperator}`
+    | RelationshipWhereOperator;
+
+export const comparisonMap: Record<Exclude<WhereOperator, RelationshipWhereOperator>, string> = {
+    NOT: "=",
+    // Numerical
+    GT: ">",
+    GTE: ">=",
+    LT: "<",
+    LTE: "<=",
+    // Distance
+    DISTANCE: "=",
+    // String
+    NOT_CONTAINS: "CONTAINS",
+    CONTAINS: "CONTAINS",
+    NOT_STARTS_WITH: "STARTS WITH",
+    STARTS_WITH: "STARTS WITH",
+    NOT_ENDS_WITH: "ENDS WITH",
+    ENDS_WITH: "ENDS WITH",
+    // Regex
+    MATCHES: "=~",
+    // Array
+    NOT_IN: "IN",
+    IN: "IN",
+    NOT_INCLUDES: "IN",
+    INCLUDES: "IN",
+};
+
+export const whereRegEx =
+    /(?<fieldName>[_A-Za-z]\w*?)(?<isAggregate>Aggregate)?(?:_(?<operator>NOT|NOT_IN|IN|NOT_INCLUDES|INCLUDES|MATCHES|NOT_CONTAINS|CONTAINS|NOT_STARTS_WITH|STARTS_WITH|NOT_ENDS_WITH|ENDS_WITH|LT|LTE|GT|GTE|DISTANCE|ALL|NONE|SINGLE|SOME))?$/;
+export type WhereRegexGroups = {
+    fieldName: string;
+    isAggregate?: string;
+    operator?: WhereOperator;
+};
+
+export const createWhereClause = ({
```

Comment:
**suggestion (non-blocking):** This seems to be an extension of `createFilter` which previously existed, and also appears to be quite a unit testable function, but doesn't appear to have any tests? Even though we have the TCK tests for this, it would be super cool to see some unit tests validating the output of this function!

Reply:
Added unit tests in https://github.com/neo4j/graphql/pull/796/commits/74a18321c7a5c5d4b84a178cf6ffd15467a4985d